test/jdk/javax/swing/JViewport/bug4546474.java

```diff
@@ -28,46 +28,31 @@
  * @run main bug4546474
  */
 
-import java.awt.Dimension;
-import java.awt.Robot;
-
 import javax.swing.JPanel;
 import javax.swing.JScrollBar;
 import javax.swing.JScrollPane;
-import javax.swing.SwingUtilities;
 
 public class bug4546474 {
     static JScrollPane scrollpane;
     static JScrollBar sbar;
     static volatile boolean viewChanged;
 
-    public static void main(String[] args) throws Exception {
-        SwingUtilities.invokeAndWait(() -> {
-            JPanel panel = new JPanel();
-            panel.setPreferredSize(new Dimension(500, 500));
-            scrollpane = new JScrollPane(panel,
-                    JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
-                    JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
-            sbar = scrollpane.getVerticalScrollBar();
-        });
+    public static void main(String[] args) {
+        JPanel panel = new JPanel();
+        scrollpane = new JScrollPane(panel,
+                JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        sbar = scrollpane.getVerticalScrollBar();
 
-        Robot robot = new Robot();
-        robot.delay(500);
-        SwingUtilities.invokeAndWait(() -> {
-            sbar.addAdjustmentListener(e -> viewChanged = true);
-            scrollpane.setViewportView(null);
-        });
-        robot.delay(500);
+        sbar.addAdjustmentListener(e -> viewChanged = true);
+        scrollpane.setViewportView(null);
         if (!viewChanged) {
             viewChanged = true;
         }
```

Comment:
What purpose does viewChanged have ?
Its set but never read.

Reply:
The adjustmentListener therefore seems irrelevant too.

Reply:
Seems to me the test reduces to
import javax.swing.JPanel;
import javax.swing.JScrollBar;
import javax.swing.JScrollPane;

public class bug4546474 {
    static JScrollPane scrollpane;
    static JScrollBar sbar;

    public static void main(String[] args) {
        JPanel panel = new JPanel();
        scrollpane = new JScrollPane(panel,
                JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
        sbar = scrollpane.getVerticalScrollBar();
        scrollpane.setViewportView(null);

        if (sbar.getVisibleAmount() > 0) {
            throw new RuntimeException("Vertical scrollbar is not " +
                    "updated when viewport is replaced");
        }
    }
}

Reply:
Updated similarly. Thanks